modules/recording/recording/template-detail.tsx

```diff
@@ -0,0 +1,294 @@
+import { GlobalCore } from "@/core/module/module.types";
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import templateService, {
+  Template,
+  TemplateField,
+} from "@/services/genesis/templates.service";
+import template_styles from "./styles/template-detail.module.css";
+import { FaEdit, FaSave, FaTrash, FaArrowLeft, FaPlus } from "react-icons/fa";
+import MessageHandler from "@/core/message-handler";
+import DeleteConfirmation from "@/resources/containers/delete-confirmation";
+import { useTranslation } from "react-i18next";
+
+const messageHandler = MessageHandler.get();
+
+const TemplateDetail = () => {
+  const router = useRouter();
+  const { t } = useTranslation();
+  const { id } = router.query;
+  const [template, setTemplate] = useState<Template | null>(null);
+  const [editingField, setEditingField] = useState<string | null>(null);
+  const [editedValues, setEditedValues] = useState<{
+    [key: string]: { name: string } & TemplateField;
+  }>({});
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  const [fieldToDelete, setFieldToDelete] = useState<string | null>(null);
+
+  const typeOptions = [
+    "string",
+    "number",
+    "boolean",
+    "object",
+    "array",
+    "date",
+  ];
+
+  useEffect(() => {
+    if (id) {
+      fetchTemplate();
+    }
+  }, [id]);
+
+  const fetchTemplate = async () => {
+    if (typeof id === "string") {
+      try {
+        const fetchedTemplate = await templateService.getTemplate(parseInt(id));
+        if (fetchedTemplate) {
+          setTemplate(fetchedTemplate);
+        } else {
+          messageHandler.handleError("Template not found");
+        }
+      } catch (error) {
+        messageHandler.handleError("Failed to fetch template");
+      }
```

Comment:
Keep the try-tach inside the service if required, not outside, also the error managment